pkg/sql/opt/exec/execbuilder/mutation.go

```diff
@@ -51,6 +51,37 @@ func (b *Builder) buildMutationInput(
 	return input, nil
 }
 
+func (b *Builder) buildInsertInput(ins *memo.InsertExpr, colList opt.ColList) (execPlan, error) {
+	// Unlike with the other three mutation expressions, it never makes sense
+	// to apply implicit row-level locking to the input of an INSERT expression
+	// because any contention results in unique constraint violations.
+	return b.buildMutationInput(ins.Input, colList, &ins.MutationPrivate)
+}
+
+func (b *Builder) buildUpdateInput(upd *memo.UpdateExpr, colList opt.ColList) (execPlan, error) {
```

Comment:
[nit] Instead of these variants, you could have a shouldApplyImplicitLockingToMutationInput which takes the mutation op as an argument and does a switch on it inside.

Reply:
Let me know if you like that better. Added as a separate commit and will squash if you give it a 👍

Reply:
Friendly ping on this. Happy to go with whatever you prefer.